domain/application/state/application.go

```diff
@@ -1219,23 +1210,22 @@ func (st *State) insertCloudServiceAddresses(
 		return nil
 	}
 
-	subnetUUID, err := st.k8sSubnetUUID(ctx, tx)
-	if err != nil {
-		return errors.Capture(err)
-	}
-
 	ipAddresses := make([]ipAddress, len(addresses))
 	for i, address := range addresses {
 		// Create a UUID for new addresses.
 		addrUUID, err := uuid.NewUUID()
 		if err != nil {
 			return errors.Capture(err)
 		}
+		cidr, err := cidrForCloudServiceIP(address.AddressType())
+		if err != nil {
+			return errors.Capture(err)
+		}
 		ipAddresses[i] = ipAddress{
```

Comment:
We need to move this to the network domain, which already has this type and encoding methods.